sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/planning/patterns.scala

```diff
@@ -92,6 +92,36 @@ object PhysicalOperation extends PredicateHelper {
           .map(Alias(_, a.name)(a.exprId, a.qualifier, isGenerated = a.isGenerated)).getOrElse(a)
     }
   }
+
+  /**
+   * Drop the non-partition key expression in the disjunctions, to optimize the partition pruning.
```

Comment:
"Drop the non-partition key expression in the disjunctions". Should it be "conjunctions"?